From d3060fdd91cefbae1a60ac90872311b239c556c2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:51 -0400
Subject: [powerpc] tm: Use STK_PARAM

Message-id: <1410545655-205645-423-git-send-email-dzickus@redhat.com>
Patchwork-id: 94363
O-Subject: [RHEL7 PATCH 422/626] powerpc/tm: Use STK_PARAM
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit c2e31bdc120d992a90b75d94c7fa403fea362e43
Author: Anton Blanchard <anton@samba.org>
Date:   Mon Mar 10 10:48:44 2014 +1100

    powerpc/tm: Use STK_PARAM

    Get rid of the tm specific STACK_PARAM and use STK_PARAM

    Signed-off-by: Anton Blanchard <anton@samba.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/tm.S b/arch/powerpc/kernel/tm.S
index 5752ace..77206d7 100644
--- a/arch/powerpc/kernel/tm.S
+++ b/arch/powerpc/kernel/tm.S
@@ -41,7 +41,6 @@ END_FTR_SECTION_IFSET(CPU_FTR_VSX);  \
 /* Stack frame offsets for local variables. */
 #define TM_FRAME_L0 TM_FRAME_SIZE-16
 #define TM_FRAME_L1 TM_FRAME_SIZE-8
-#define STACK_PARAM(x) (48+((x)*8))
 
 
 /* In order to access the TM SPRs, TM must be enabled.  So, do so: */
@@ -113,7 +112,7 @@ _GLOBAL(tm_reclaim)
 
  /* We've a struct pt_regs at [r1+STACK_FRAME_OVERHEAD]. */
 
- std r3, STACK_PARAM(0)(r1)
+ std r3, STK_PARAM(R3)(r1)
  SAVE_NVGPRS(r1)
 
  /* We need to setup MSR for VSX register save instructions.  Here we
@@ -202,7 +201,7 @@ dont_backup_fp:
  /* Now get some more GPRS free */
  std r7, GPR7(r1)   /* Temporary stash */
  std r12, GPR12(r1)   /* ''   ''    ''   */
- ld r12, STACK_PARAM(0)(r1)  /* Param 0, thread_struct * */
+ ld r12, STK_PARAM(R3)(r1)  /* Param 0, thread_struct * */
 
  std r11, THREAD_TM_PPR(r12)  /* Store PPR and free r11 */
 
-- 
1.7.1